File: src/Servers/QueryReport/src/Aggregate/Redis/PeerRoomInfo.cs
using System;
using System.Collections.Generic;


//!fix move this to root namespace 
namespace UniSpy.Server.QueryReport.Aggregate.Redis.PeerGroup
{
    public record PeerRoomInfo
    {
        public Guid? ServerId { get; set; }
        public string GameName { get; private set; }
        public int? GroupId { get; private set; }
        public string RoomName { get; private set; }
        public int NumberOfWaitingPlayers
        {
            get => int.Parse(KeyValues["numwaiting"]);
            set => KeyValues["numwaiting"] = value.ToString();
        }
        public int MaxNumberOfWaitingPlayers
        {
            get => int.Parse(KeyValues["maxwaiting"]);
            set => KeyValues["maxwaiting"] = value.ToString();
        }
        public int NumberOfServers
        {
            get => int.Parse(KeyValues["numservers"]);
            set => KeyValues["numservers"] = value.ToString();
        }

        public int MaxNumberOfPlayers
        {
            get => int.Parse(KeyValues["maxplayers"]);
            set => KeyValues["maxplayers"] = value.ToString();
        }
        public string Password => (string)KeyValues["password"];
        public int NumberOfGames
        {
            get => int.Parse(KeyValues["numgames"]);
            set => KeyValues["numgames"] = value.ToString();
        }
        public int NumberOfPlayers
        {
            get => int.Parse(KeyValues["numplayers"]);
            set => KeyValues["numplayers"] = value.ToString();
        }
        public int NumberOfPlayingPlayers
        {
            get => int.Parse(KeyValues["numplaying"]);
            set => KeyValues["numplaying"] = value.ToString();
        }
        public Dictionary<string, string> KeyValues { get; private set; } = new Dictionary<string, string>();
        public PeerRoomInfo(string gameName, int groupId, string roomName)
        {
            // this is the default value that every game needed
            GameName = gameName;
            GroupId = groupId;
            RoomName = roomName;
            KeyValues.Add("groupid", groupId.ToString());
            KeyValues.Add("hostname", roomName);
            KeyValues.Add("numwaiting", "0");
            KeyValues.Add("maxwaiting", "200");
            KeyValues.Add("numservers", "0");
            KeyValues.Add("numplayers", "0");
            KeyValues.Add("maxplayers", "200");
            KeyValues.Add("password", "");
            KeyValues.Add("numgames", "0");
            KeyValues.Add("numplaying", "0");
        }
    }
}

File: src/servers/query_report/src/aggregates/peer_room_info.py
from uuid import UUID

from pydantic import BaseModel


dd = {
    "groupid": "groupid",
    "hostname": "hostname",
    "number_of_waiting_player": "numwaiting",
    "max_number_of_waiting_players": "maxwaiting",
    "number_of_servers": "numservers",
    "number_of_players": "numplayers",
    "max_number_of_players": "maxplayers",
    "password": "password",
    "number_of_games": "numgames",
    "number_of_playing_players": "numplaying",
}


class PeerRoomInfo(BaseModel):
    server_id: UUID
    game_name: str
    group_id: int
    room_name: str
    # todo change to dict[str, object]
    raw_key_values: dict

    def __init__(self, game_name, group_id, room_name) -> None:
        self.game_name = game_name
        self.group_id = group_id
        self.room_name = room_name
        self.raw_key_values = {
            "groupid": group_id,
            "hostname": room_name,
            "numwaiting": 0,
            "maxwaiting": 200,
            "numservers": 0,
            "numplayers": 0,
            "maxplayers": 200,
            "password": "",
            "numgames": 0,
            "numplaying": 0,
        }

    @property
    def number_of_servers(self) -> int:
        return int(self.raw_key_values["numservers"])

    @number_of_servers.setter
    def number_of_servers(self, value: int):
        assert isinstance(value, int)
        self.raw_key_values["numservers"] = value

    @property
    def number_of_players(self) -> int:
        return int(self.raw_key_values["numplayers"])

    @number_of_players.setter
    def number_of_players(self, value: int):
        assert isinstance(value, int)
        self.raw_key_values["numplayers"] = value

    @property
    def max_number_of_waiting_players(self) -> int:
        return int(self.raw_key_values["maxwaiting"])

    @max_number_of_waiting_players.setter
    def max_number_of_waiting_players(self, value: int):
        assert isinstance(value, int)
        self.raw_key_values["maxwaiting"] = value

    @property
    def max_number_of_players(self) -> int:
        return int(self.raw_key_values["maxplayers"])

    @max_number_of_players.setter
    def max_number_of_players(self, value: int):
        assert isinstance(value, int)
        self.raw_key_values["maxplayers"] = value

    @property
    def number_of_games(self) -> int:
        return int(self.raw_key_values["numgames"])

    @number_of_games.setter
    def number_of_games(self, value: int):
        assert isinstance(value, int)
        self.raw_key_values["numgames"] = value

    @property
    def number_of_playing_players(self) -> int:
        return self.raw_key_values["numplaying"]

    @number_of_playing_players.setter
    def number_of_playing_players(self, value: int):
        assert isinstance(value, int)
        self.raw_key_values["numplaying"] = value

    def get_gamespy_format_data(self):
        """
        convert everything to string
        """
        return {key: str(value) for key, value in self.raw_key_values.items()}
